x-pack/plugins/apm/public/components/app/service_overview/service_overview_transactions_table/index.tsx

```diff
@@ -124,140 +79,97 @@ export function ServiceOverviewTransactionsTable(props: Props) {
           start,
           end,
           uiFilters: JSON.stringify(uiFilters),
-          size: PAGE_SIZE,
-          numBuckets: 20,
-          pageIndex: tableOptions.pageIndex,
-          sortField: tableOptions.sort.field,
-          sortDirection: tableOptions.sort.direction,
           transactionType,
           latencyAggregationType: latencyAggregationType as LatencyAggregationType,
         },
       },
-    }).then((response) => {
-      return {
-        items: response.transactionGroups,
-        totalItemCount: response.totalTransactionGroups,
-        tableOptions: {
-          pageIndex: tableOptions.pageIndex,
-          sort: {
-            field: tableOptions.sort.field,
-            direction: tableOptions.sort.direction,
-          },
-        },
-      };
     });
   }, [
     serviceName,
     start,
     end,
     uiFilters,
-    tableOptions.pageIndex,
-    tableOptions.sort.field,
-    tableOptions.sort.direction,
     transactionType,
     latencyAggregationType,
   ]);
 
+  const { transactionGroups, requestId } = data;
+  const sortedSlicedItems = orderBy(
```

Comment:
nit: instead of naming it after the implementation, perhaps name it after what it is
```suggestion
  const currentPageTransactionGroups = orderBy(
```